include/pros/imu.h

```diff
@@ -0,0 +1,295 @@
+/**
+ * \file pros/imu.h
+ *
+ * Contains prototypes for functions related to the VEX Inertial sensor.
+ *
+ * Visit https://pros.cs.purdue.edu/v5/tutorials/topical/imu.html to learn
+ * more.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2019, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef _PROS_IMU_H_
+#define _PROS_IMU_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+namespace pros {
+namespace c {
+#endif
+
+typedef enum imu_status_e {
+	E_IMU_STATUS_CALIBRATING = 0x01,
+	E_IMU_STATUS_ERROR = 0xFF,  // NOTE: used for returning an error from the get_status function, not that the IMU is
+	                            // necessarily in an error state
+} imu_status_e_t;
+
+typedef struct __attribute__((__packed__)) quaternion_s {
+	double x;
+	double y;
+	double z;
+	double w;
+} quaternion_s_t;
+
+struct __attribute__((__packed__)) imu_raw_s {

```

Comment:
The layout would probably be best left to the compiler, since it's not used for interop with vexos.

Reply:
That's fair. I tossed that around in my head a bit, and eventually figured I'd just pack it (e.g. in case someone wants to send it directly over a wire, or something). Not terribly committed either way, so I'll get rid of that